### PR TITLE
REST API and CRUD

### PR DESCRIPTION
**Pull request by shirrotem**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const MONGO_URL =
+  "mongodb+srv://admin:admin@cluster0.htcnfoy.mongodb.net/?retryWrites=true&w=majority";
+mongoose.Promise = Promise;
+mongoose.connect(MONGO_URL);
+mongoose.connection.on("error", (error: Error) => console.log(error));
+
+app.use("/", router());
+app.get("/", (req: Request, res: Response) => {
+  res.send("Welcome to the default route!");
```
Comment (MoranOshiaMoveo) [inline src/index.ts]: change to meaning-full log

File: src/routes/chefRoute.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    chef.name = name;
+    await chef.save();
+
+    return res.status(200).json(chef);
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: "Internal Server Error" });
+  }
+};
```
Comment (MoranOshiaMoveo) [inline src/routes/chefRoute.ts]: all this func is controllers , please open controllers folder and add there file for each type and there put the controller file instead in the route file

File: src/routes/dishRoute.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    dish.name = name;
+    await dish.save();
+
+    return res.status(200).json(dish);
+  } catch (error) {
+    console.log(error);
+    return res.sendStatus(500);
+  }
+};
```
Comment (MoranOshiaMoveo) [inline src/routes/dishRoute.ts]: same as abive

File: src/routes/restaurantRoute.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    restaurant.name = name;
+    await restaurant.save();
+
+    return res.status(200).json(restaurant).end();
+  } catch (error) {
+    console.log(error);
+    return res.sendStatus(400);
+  }
+};
```
Comment (MoranOshiaMoveo) [inline src/routes/restaurantRoute.ts]: same as above

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (shirrotem) [reply src/index.ts]: fixed

File: src/routes/chefRoute.ts (hunk 2 above)
Comment (shirrotem) [reply src/routes/chefRoute.ts]: fixed

File: src/routes/dishRoute.ts (hunk 3 above)
Comment (shirrotem) [reply src/routes/dishRoute.ts]: fixed

File: src/routes/restaurantRoute.ts (hunk 4 above)
Comment (shirrotem) [reply src/routes/restaurantRoute.ts]: fixed

**Review by MoranOshiaMoveo — APPROVED** [review]

**Pull request merged**